### PR TITLE
[INT-13638] Add Dialog and Inline components

### PR DESCRIPTION
**Pull request by andrewhillcode**
INT-13638

**The following MUST be checked prior to approval. If not relevant to your PR, remove the line from your description.**
- [ ]  The `styleguide.config.js` has been updated to show examples/new functionality for the component
- [ ]  Test Coverage for any new or updated functionality
- [ ]  New and updated components have been tested locally in lapis and SPA and work as expected
- [ ]  This PR has been tagged with PATCH, MINOR, or MAJOR.
- [ ]  The component has data-component-type and data-component-context attributes following the standard
- [ ]  You have requested a cross-team review on this component so everyone knows it exists

### CODE REVIEW

File: src/domain/Layouts/Inline/Inline.examples.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+The **Inline** is a compositional component that will space each child element an equal distance apart.
+The align prop can be used to align items to the left, center, or right of a box.
+
+```jsx
```
Comment (wyvl) [inline src/domain/Layouts/Inline/Inline.examples.md:4]: add example with button

File: src/domain/Layouts/Inline/Inline.tsx [hunk 2]
```diff
@@ -0,0 +1,50 @@
+import React from 'react'
+
+import { Props, Variables } from '../../../common'
+import { StyledInlineElement, StyledInlineGroup, StyledInlineGroupWrapper } from './style'
+
+interface IInlineProps {
```
Comment (wyvl) [inline src/domain/Layouts/Inline/Inline.tsx:6]: should we have collapseBelow props too? It is highly used in out system. We are using the grid layout for full-width mobile right now.

File: src/domain/Layouts/Inline/Inline.tsx [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+      <StyledInlineGroup
+        spacing={spacing}
+        align={align}
+      >
+        {wrappedChildren}
+      </StyledInlineGroup>
+    </StyledInlineGroupWrapper>
+  )
+}
+
```
Comment (wyvl) [inline src/domain/Layouts/Inline/Inline.tsx:45]: InLine.Align = InlineAlign

File: src/domain/Layouts/Inline/Inline.tsx (hunk 2 above)
Comment (wyvl) [inline src/domain/Layouts/Inline/Inline.tsx:6]: margins props

File: src/domain/Layouts/Inline/style.ts [hunk 4]
```diff
@@ -0,0 +1,65 @@
+import styled, { css } from 'styled-components'
+
+import { InlineAlign } from './Inline'
+
```
Comment (wyvl) [inline src/domain/Layouts/Inline/style.ts:4]: interface IStyledInlineGroupWrapperProps {
  margins?: Props.IMargins
spacing?: Variables.Spacing | Variables.Layout
}

File: src/domain/Modals/Dialog/Dialog.examples.md [hunk 5]
```diff
@@ -0,0 +1,68 @@
+**Dialogs** are modals that are intended to be use only for quick binary actions like:     
+Yes or No / Delete or Cancel
+
+They come in two variants: `positive` and `negative`.    
+The variant should be chosen based on the sentiment of the primary action.
+
```
Comment (wyvl) [inline src/domain/Modals/Dialog/Dialog.examples.md:6]: add "Actions", "Best Practices", "Related Components" (could include "Differences from a Modal") session

File: src/domain/Modals/Dialog/Dialog.tsx [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+  title: string
+  onPrimaryActionClick: (event: (React.MouseEvent<HTMLElement, MouseEvent>)) => void
+  primaryActionLabel: string
+  onSecondaryActionClick?: (event: (React.MouseEvent<HTMLElement, MouseEvent>)) => void
+  secondaryActionLabel?: string
+  variant: DialogVariant
+  componentContext: string
+}
+
+enum DialogVariant {
```
Comment (wyvl) [inline src/domain/Modals/Dialog/Dialog.tsx:23]: should it be neutral and alert?

File: src/domain/Modals/Dialog/style.tsx [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+    height: auto;
+    margin: auto;
+    outline: none;
+    overflow-y: visible;
+    padding: ${dialogPadding}px;
+    position: relative;
+    max-width: ${dialogContentMaxWidth}px;
+
+    ${Utils.mediaQueryBetweenSizes({ maxPx: totalDialogWidth })} {
+      margin: auto 24px;
```
Comment (wyvl) [inline src/domain/Modals/Dialog/style.tsx:81]: use Variables

File: src/domain/Layouts/Inline/Inline.tsx (hunk 3 above)
Comment (andrewhillcode) [reply src/domain/Layouts/Inline/Inline.tsx:45]: does that work? i couldnt find any examples of it being used elsewhere

File: src/domain/Modals/Dialog/Dialog.tsx (hunk 6 above)
Comment (andrewhillcode) [reply src/domain/Modals/Dialog/Dialog.tsx:23]: I think the names can change depending on the context of the component. Like one colour can have many names:
green = approved, confirmed, success
red = noncompliant, declined, error

File: src/domain/Layouts/Inline/Inline.tsx (hunk 2 above)
Comment (andrewhillcode) [reply src/domain/Layouts/Inline/Inline.tsx:6]: I'm not sure. CollapseBelow doesnt work in cases like goal template (unless we don't want to do that anymore...)

File: src/domain/Modals/Dialog/Dialog.tsx (hunk 6 above)
Comment (andrewhillcode) [reply src/domain/Modals/Dialog/Dialog.tsx:23]: In this case you're either performing a positive (blue) action or a negative (red) action

File: src/domain/Layouts/Inline/Inline.tsx (hunk 2 above)
Comment (andrewhillcode) [reply src/domain/Layouts/Inline/Inline.tsx:6]: We will support flex grow and collapse below

File: src/domain/Layouts/Inline/Inline.tsx [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+  children
+}) => {
+  const wrappedChildren = React.Children.map(
+    children,
+    (child: any) => {
+      let childProps = {}
+
+      if (child && child?.type && child?.type?.name === Item.name) {
+        childProps = child?.props
+      }
```
Comment (andrewhillcode) [inline src/domain/Layouts/Inline/Inline.tsx:37]: This is definitely kind of weird but I think it provides the easiest experience when actually using the component (magically being able to wrap anything in `<Inline.Item fill>` and have it work. No clue how to get the typing to work (well I kind of do but I think there might be a better way) so I've just left it as `any` for now.

File: src/domain/Layouts/Inline/Inline.examples.md [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
```
Comment (wyvl) [inline src/domain/Layouts/Inline/Inline.examples.md:4]: add example with collapse true

File: src/domain/Layouts/Inline/Inline.tsx (hunk 8 above)
Comment (wyvl) [reply src/domain/Layouts/Inline/Inline.tsx:37]: may be add an interface for InlineItem?

File: src/domain/Modals/Dialog/Dialog.examples.md [hunk 10]
```diff
@@ -0,0 +1,84 @@
...
+<br />
+
+### Best Practices
+* For actions, avoid 'Yes' and 'No' responses.
+* If an action involves cancelling an operation, it should be the secondary action, usually labeled 'Cancel'.
+
+<br />
+
+### Related Components
+* For more complex interactions that require user interruption, use a [Modal](/#/Overlay/Modal).
```
Comment (wyvl) [inline src/domain/Modals/Dialog/Dialog.examples.md:84]: may be talk about different between modal and dialog like:
dialog can be on top of modal while modal can't have extra modal on top

Comment (wyvl) [reply src/domain/Modals/Dialog/Dialog.examples.md:84]: Modal will be full page on mobile but dialog will not

File: src/domain/Layouts/Inline/Inline.examples.md (hunk 9 above)
Comment (andrewhillcode) [reply src/domain/Layouts/Inline/Inline.examples.md:4]: too many examples :(

**Pull request merged**